lib/main.dart

```diff
@@ -13,15 +13,18 @@ import 'package:trash_themes/themes.dart';
 Future<void> main() async {
   var provider = await AppProvider().init();
 
+  //FIXME: put the setting inside the APIProvider
   Setting setting = Setting();
 
   await getSettingsInfo().then((value) => {
         setting = value,
       });
 
-  if (setting.path != "No path found") {
+  // FIXME: add a method isValid() inside the setting to check if all the configuration are
+  // correct, otheriwise throws an error to disply to the user
+  if (setting.path != "No path found" || setting.clientMode == clients[2]) {
     provider.registerLazyDependence<AppApi>(() {
-      if (setting.connectionType == clients[0]) {
+      if (setting.clientMode == clients[0]) {
```

Comment:
we could migrate also this code, and we can avoid using `client[1]` but we can simply move the architecture to use the setting and we save a couple of code